src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestworker/services/GeneratePdfServiceRestrictedPatientsTest.kt

```diff
@@ -0,0 +1,83 @@
+package uk.gov.justice.digital.hmpps.hmppssubjectaccessrequestworker.services
+
+import com.itextpdf.kernel.pdf.PdfDocument
+import com.itextpdf.kernel.pdf.PdfReader
+import com.itextpdf.kernel.pdf.PdfWriter
+import com.itextpdf.kernel.pdf.canvas.parser.PdfTextExtractor
+import com.itextpdf.layout.Document
+import io.kotest.core.spec.style.DescribeSpec
+import org.assertj.core.api.Assertions.assertThat
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.ConfigDataApplicationContextInitializer
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import uk.gov.justice.digital.hmpps.hmppssubjectaccessrequestworker.models.DpsService
+import java.io.FileOutputStream
+
+@ActiveProfiles("test")
+@ContextConfiguration(
+  initializers = [ConfigDataApplicationContextInitializer::class],
+  classes = [(GeneratePdfService::class)],
+)
+class GeneratePdfServiceRestrictedPatientsTest(
+  @Autowired val generatePdfService: GeneratePdfService,
+) : DescribeSpec(
+  {
+    fun writeAndThenReadPdf(
+      testInput: Map<String, String>?,
+    ): PdfDocument {
+      val testFileName = "dummy-template-restricted-patients.pdf"
+      val testResponseObject = listOf(DpsService(name = "hmpps-restricted-patients-api", content = testInput))
+      val mockPdfDocument = PdfDocument(PdfWriter(FileOutputStream(testFileName)))
```

Comment:
seems a bit naff that these `dummy-*` files are all written to the root of the project rather than to a junit temp test directory

Reply:
Good point